pkg/infra/teams/team3/loot_strat.go

```diff
@@ -92,13 +92,10 @@ func (a *AgentThree) HandleLootInformation(m message.TaggedInformMessage[message
 
 // forcibly call at start of loot phase to begin proceedings
 func (a *AgentThree) RequestLootProposal(baseAgent agent.BaseAgent) { // put your logic here now, instead
-	a.mutex.Lock()
 	sendProposal := rand.Intn(100)
 	if sendProposal > a.personality {
-		a.mutex.Unlock()
 		return
 	}
-	a.mutex.Unlock()
```

Comment:
I'm pretty sure we need this one - we get a race condition on r/w to `a.personality`

Reply:
Since there’s no goroutines launched from the agent itself — I.e. each agent only has one thread of execution concurrently, I disagree and the race detector doesn’t detect any races here either even with massive scale.

Reply:
From multiple runs with the race detector, we have seen that the threading of `UpdateInternalState` may coincide with the threading of `HandleElection` based on the sleep duration. Both functions access the map `a.personality` causing concurrent read and writes

Reply:
I would advise that you run the race detector without these mutexes across multiple iterations

Reply:
After you mentioned `HandleElection`, the root cause of the issue can be found [here](https://github.com/antonypap/SOMAS2022/blob/6268965b7acb27ba1e9a1fe6c9aaaf8330f56b82/pkg/infra/game/stage/election/election.go#L48-L55), I've patched in the latest commit and will rebase.